City-of-Helsinki/drupal-hdbt
# UHF-2755 main nav accessibility

# [UHF-2755](https://helsinkisolutionoffice.atlassian.net/browse/UHF-2755) Main nav accessibility

## What was done

### Main navigation

* Added accessible name to nav chevron buttons using `aria-labelledby`
  * 4241be0
* Removed duplicate id attributes from navigation and elsewhere
  * 5ba831b 
  * 6a06ec7
  * 92724f5
  * ce64a7a
* Added `aria-haspopup` attribute to main nav dropdown buttons
  * 6a843c3
* Added `aria-current=page` attribute to all links pointing to current page
  * d7ed0d6
  * b898396
* Added explanation text for all external link icons on site except ones that are added via ckeditor.
  * 3df6c21
  * 357a05a
  * 6648ed5
  * dd8941e
  * 14ed6a2
  * 5eeff7d

### Mobile navigation dialog

* Added `role=button` to menu hamburger button as it's not meant to be a "link" that takes you to another location, but opens something like a button instead.
  * 9cdb28b
* Added `aria-haspopup=dialog` to menu hamburger button to indicate that it opens a modal dialog
  * 9dbd02d
* Added `role=dialog` and `aria-modal=true` to mobile menu dialog
  * e56c9dd

### Other fixes on the side

* Fixed table text line-heights
  * 1154777
* Removed branding menu submenu code as it's not meant to be used and was broken
  * 076bd0f
* Fixed cookie consent dialog on small screens (max-height:100vh and overflow:auto)
  * ce377f3
* Removed duplicate (or quadruplicate) code from menu templates
  * 9bb6987
  * 87c5dde

## How to test

* Get branch
* Make sure you have latest helfi_media_map module (`composer require drupal/helfi_media_map:^1.0` or ask Lari)
* Open or create a page that has:
  * Top navigation
  * Branding navigation
  * Main navigation first level
  * Main navigation second level
  * Side navigation all available levels
  * Footer top
  * Footer bottom
  * Map with an external link icon
* Check these for added attributes, external link help texts and other fixes mentioned above
  * On desktop
  * On mobile
 
![image](https://user-images.githubusercontent.com/1191667/142379022-cc3b4247-bc22-4bac-8a46-5470a9f133f7.png)
